`test-curl` tests fails in dev envoriments

### Description

About half of the test_curl tests fail when running in a dev environment - because the dev environment does not have a x509 cert the way these tests expect. 

```
FAILED tests/test_curl.py::TestCurlRucio::test_get_auth_x509 - AssertionError: assert 'X-Rucio-Auth-Token' in ''
FAILED tests/test_curl.py::TestCurlRucio::test_get_auth_validate - AssertionError: assert 'X-Rucio-Auth-Token' in ''
FAILED tests/test_curl.py::TestCurlRucio::test_post_account - AssertionError: assert 'X-Rucio-Auth-Token' in ''
FAILED tests/test_curl.py::TestCurlRucio::test_get_accounts_whoami - AssertionError: assert 'X-Rucio-Auth-Token' in ''
FAILED tests/test_curl.py::TestCurlRucio::test_post_rse - AssertionError: assert 'X-Rucio-Auth-Token' in ''
```

### Steps to reproduce

- install the dev environment using [the instructions here](https://github.com/rucio/rucio/tree/master/etc/docker/dev)
- Run either `bash tools/run_tests.sh` or `pytest tests/test_curl.py` - observe the failed tests

### Rucio Version

_No response_

### Additional Information

I propose we mark these tests with a `skip_if` to only run them in the github runner or a specific environment.

--- a/lib/rucio/tests/common.py
+++ b/lib/rucio/tests/common.py
@@ -45,6 +45,8 @@ skip_multivo = pytest.mark.skipif('SUITE' in os.environ and os.environ['SUITE'] 
                                   reason="does not work for multiVO")
 skip_non_belleii = pytest.mark.skipif(not ('POLICY' in os.environ and os.environ['POLICY'] == 'belleii'),
                                       reason="specific belleii tests")
+skip_outside_gh_actions = pytest.mark.skipif(os.getenv("GITHUB_ACTIONS") != "true",
+                                             reason="Skipping tests outside GitHub Actions")
 
 
 def is_influxdb_available() -> bool:

--- a/tests/test_curl.py
+++ b/tests/test_curl.py
@@ -18,7 +18,7 @@ import os
 import pytest
 
 from rucio.common.config import config_get, config_get_bool
-from rucio.tests.common import account_name_generator, execute, get_long_vo, rse_name_generator
+from rucio.tests.common import account_name_generator, execute, get_long_vo, rse_name_generator, skip_outside_gh_actions
 
 
 class TestCurlRucio:
@@ -54,6 +54,7 @@ class TestCurlRucio:
         print(out, )
         assert 'X-Rucio-Auth-Token' in out
 
+    @skip_outside_gh_actions
     def test_get_auth_x509(self):
         """AUTH (CURL): Test auth token retrieval with via x509"""
         cmd = 'curl -s -i --cacert %s -H "X-Rucio-Account: root" %s -cert %s --key %s -X GET %s/auth/x509' % (self.cacert, self.vo_header, self.usercert, self.userkey, self.auth_host)
@@ -86,6 +87,7 @@ class TestCurlRucio:
         print(out, )
         # assert 'X-Rucio-Auth-Token' in out
 
+    @skip_outside_gh_actions
     def test_get_auth_validate(self):
         """AUTH (CURL): Test if token is valid"""
         cmd = 'curl -s -i --cacert %s -H "X-Rucio-Account: root" %s --cert %s --key %s -X GET %s/auth/x509 | tr -d \'\r\' | grep X-Rucio-Auth-Token:' % (self.cacert, self.vo_header, self.usercert, self.userkey, self.auth_host)
@@ -98,6 +100,7 @@ class TestCurlRucio:
         print(out)
         assert 'datetime.datetime' in out
 
+    @skip_outside_gh_actions
     @pytest.mark.dirty
     def test_post_account(self):
         """ACCOUNT (CURL): add account"""
@@ -111,6 +114,7 @@ class TestCurlRucio:
         print(out)
         assert '201 Created'.lower() in out.lower()
 
+    @skip_outside_gh_actions
     def test_get_accounts_whoami(self):
         """ACCOUNT (CURL): Test whoami method"""
         cmd = 'curl -s -i --cacert %s -H "X-Rucio-Account: root" %s --cert %s --key %s -X GET %s/auth/x509 | tr -d \'\r\' | grep X-Rucio-Auth-Token:' % (self.cacert, self.vo_header, self.usercert, self.userkey, self.auth_host)
@@ -124,6 +128,7 @@ class TestCurlRucio:
         print(out)
         assert '303 See Other'.lower() in out.lower()
 
+    @skip_outside_gh_actions
     @pytest.mark.dirty
     def test_post_rse(self):
         """RSE (CURL): add RSE"""

